### DISCUSSION TITLE
Is there any documentation on TreeView?

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#11090] [Archive]
When creating a field, under `Relational`, we see `Tree View`.
Is there any documentation on that?

I ask this because I may miss something before raising an issue cos using SQLite, I get an
```
INTERNAL_SERVER_ERROR
INSERT INTO _knex_temp_alter732 SELECT * FROM test; - SQLITE_CONSTRAINT: FOREIGN KEY constraint failed"
```

May be related to #10979, as the error message creating an O2M with SQLite is the same

**Comment by joselcvarela** [comment]
Hello @Pachat 
I do believe you are misconfiguring the O2M relationship because I have just tested without any issue.
Take a look on https://docs.directus.io/configuration/relationships/#one-to-many-o2m to see how you can configure properly O2M relations.

**Comment by rijkvanzanten** [comment]
cc @erondpowell

**Comment by Pachat** [comment]
Once a tree view field is created within a collection, how to display it?
(I only see the Layout Options: Calendar, Cards, Map and Table)

**Reply by joselcvarela** [reply]
Tree View is an Interface not a Layout, that means, it's only available inside item views.
https://docs.directus.io/getting-started/glossary/#interfaces
https://docs.directus.io/getting-started/glossary/#layouts

**Comment by phanect** [comment]
This discussion is posted an year ago, but allow me adding an answer for other users from Google search results.

This page may help you:
https://docs.directus.io/cookbook/data-models/build-a-content-hierarchy.html

**Reply by erondpowell** [reply]
@phanect Thanks big time for linking. This thread is actually what inspired that recipe ☺️

**Reply by ProstoA** [reply]
Unfortunately this page is no available (

**Reply by phanect** [reply]
@ProstoA Oh... Maybe the page has been removed or moved without redirection...😭

**Reply by Pachat** [reply]
The new URL is https://docs.directus.io/app/data-model/fields/relational.html#tree-view

>The Tree View interface is only available on self-referencing (recursive) relationships. See [Many-to-Any (O2M) Relationships](https://docs.directus.io/app/data-model/relationships.html#many-to-any-m2a).
